tests/test_spark.py

```diff
@@ -318,6 +318,16 @@ def test_secrets():
         LOGGER.warn("Error when deleting secret, {}".format(r.content))
 
 
+@pytest.mark.sanity
+@pytest.mark.multspaces
+def test_cli_multiple_spaces():
+    utils.run_tests(app_url=SPARK_EXAMPLES,
+                    app_args="30",
+                    expected_output="Pi is roughly 3",
+                    app_name="/spark",
+                    args_str="--conf  spark.cores.max=2  --class   org.apache.spark.examples.SparkPi")
```

Comment:
Also, we have added an additional parameter `args_str` here just to handle this case. Would it not be better to continue to use `args` here?

We could write the parameter here as:
```python
args=["--conf", " spark.cores.max=2", " --class ", " org.apache.spark.examples.SparkPi", ]
```
(or even have the additional spaces as **explicit** empty strings in the list)
 
Since we use `" ".join(s for s in args)` in the actual submit method, there is no further manipulation fo the whitespace performed.

Reply:
There is already a unit test for this (added by Stavros in a recent PR), but Art requested an integration test as well. I'd imagine for other variants, we could have only unit tests.

Reply:
Good point about the `args_str` parameter. I'll follow your suggestion.